src/lib/support/CHIPMem-Malloc.cpp

```diff
@@ -100,30 +105,54 @@ void MemoryAllocatorShutdown()
 #endif // CHIP_CONFIG_MEMORY_DEBUG_DMALLOC
 }
 
+extern "C" {
+
+WEAK void * __wrap_malloc(size_t size)
+{
+    return malloc(size);
+}
+
+WEAK void __wrap_free(void * ptr)
+{
+    free(ptr);
+}
+
+WEAK void * __wrap_calloc(size_t num, size_t size)
+{
+    return calloc(num, size);
+}
+
+WEAK void * __wrap_realloc(void * ptr, size_t new_size)
+{
+    return realloc(ptr, new_size);
+}
+
+} // extern "C"
+
 void * MemoryAlloc(size_t size)
 {
     VERIFY_INITIALIZED();
-    return malloc(size);
+    return __wrap_malloc(size);
```

Comment:
You should only call calloc/malloc/free and not rely on __wrap* in common code. Any `__wrap*` definitions/calls should only exist under platform*. The wrapping with linker will make these calls go to wrapped versions properly without mentioning them explicitly here.

Reply:
Done modifications in the BUILD.gn file and the common code does not rely on __wrap* anymore